ipld-schema.ipldsch

```diff
@@ -0,0 +1,46 @@
+type UCAN struct {
+  version String
+
+  issuer PublicKey
+  audience PublicKey
```

Comment:
Both of these need to be strings if we want support for DIDs other than Key DID.

Reply:
> Both of these need to be strings if we want support for DIDs other than Key DID.

Could we represent other DIDs by tagging with a different multicodec code instead ?

Reply:
To support that, perhaps these fields could be `DID` instead of `PublicKey`?

Reply:
If you choose to not use String here you will basically kill interoperability. You would basically nee to specify how a huge amount of DID methods behave: https://w3c.github.io/did-spec-registries/#did-methods

I agree that byte encoding would be very nice, but you would essentially need to maintain a new registry that has a mapping function for all possible DID methods.

Reply:
I meant you could put byte representation of DID in there as opposed to string representation.

Reply:
To be more clear about my points here:

1. UCAN spec now only allows `did:key`s in issuer and audience.
2. You can put a different `did` but that is more of extension that would require additional code.
3. Likewise CBOR representation uses `Bytes` in issuer and audience.
4. You could put any bytes, could be other DID in byte representation or whatever really but just like with UCANs it would be an extension and require additional code.

I can go ahead and remove type alias and put `Bytes` instead but that is not going to change anything about points above. If we do want to support other types of DIDs we should probably have that as a separate discussion that affects UCANs overall.

Reply:
@oed does that makes more sense ? Or you do you still feel that string encoding as opposed to byte encoding affords some additional benefits that I'm failing to recognize ?

Reply:
> I meant you could put byte representation of DID in there as opposed to string representation.

Ok, that makes sense. How much space does that actually save for you though?